src/Worker.cs

```diff
@@ -79,9 +79,18 @@ public async static Task Main(string[] args)
                     }
                 });
 
+            // This must be done before the first Runspace is created, to avoid the creation of a thread by the upgrade checker
+            SetVersionUpgradeOptOut();
+
             // Create the very first Runspace so the debugger has the target to attach to.
             // This PowerShell instance is shared by the first PowerShellManager instance created in the pool,
             // and the dependency manager (used to download dependent modules if needed).
+            string checkForUpgrade = Environment.GetEnvironmentVariable("AZUREPS_CHECK_FOR_UPGRADE");
```

Comment:
Lines 88-93 seem to repeat the SetVersionUpgradeOptOut implementation. Did you mean to remove this?

Reply:
fixed